docs/reference/reranking/index.asciidoc

```diff
@@ -0,0 +1,71 @@
+[[re-ranking-overview]]
+= Re-ranking
+
+Many search systems are built on two-stage retrieval pipelines.
+
+The first stage uses cheap, fast algorithms to find a subset of possible matches.
+
+The second stage uses a more powerful model, often machine learning-based, to reorder the documents.
+This second step is called re-ranking.
+This approach balances computational costs, because the resource-intensive model is only applied to a smaller set of pre-filtered results.
+
+{es} supports various ranking and re-ranking techniques to optimize search relevance and performance.
+
+[float]
+[[re-ranking-two-stage-pipeline]]
+== Two-stage retrieval pipelines
+
+Learn about retrieval pipelines and how re-ranking can be applied to your existing search experience.
+
+[float]
+[[re-ranking-first-stage-pipeline]]
+=== First stage: initial retrieval
+
+[float]
+[[re-ranking-ranking-overview-bm25]]
+==== Full-text search: BM25 scoring
+
+{es} ranks documents based on term frequency and inverse document frequency, adjusted for document length.
+BM25 is the default statistical scoring algorithm in {es} and works out-of-the-box.
+
+[float]
+[[re-ranking-ranking-overview-vector]]
+==== Vector search: similarity scoring
+
+Vector search involves transforming data into dense or sparse vector embeddings to capture semantic meanings, and computing similarity scores for query vectors.
+Store vectors using `semantic` fields for automatic vectorization or `dense_vector` and `sparse_vector` fields when you need more control over the underlying embedding model.
+Query vectors with `semantic` queries or `knn` and `sparse_vector` queries to compute similarity scores.
+Refer to <<semantic-search,semantic search>> for more information.
+
+[float]
+[[re-ranking-ranking-overview-hybrid]]
+==== Hybrid techniques
+
+Hybrid search techniques combine results from full-text and vector search pipelines.
```

Comment:
I am not sure how much we have here, but we've been getting more and more questions and feedback on people who want more information on linear boosting as well. I wonder if a sentence here and a link to somewhere else (if we have it) would be worth doing.

Reply:
In future we want to have fuller docs sections for:

- Full-text search
- ML-powered search
- Hybrid search

I think we'll want to revisit that when the time comes. The goal here is to put LTR and semantic reranking in a smarter IA, first and foremost, not to be the exhaustive source of truth for all these topics.

Reply:
Sure sure, push back on my scope creep 😀